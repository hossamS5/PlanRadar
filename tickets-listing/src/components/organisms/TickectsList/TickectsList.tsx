import React from "react";
import TicketCard from "../../molecules/TicketCard/TicketCard";

const TickectsList: React.FC = () => {
  const list = [
    {
      id: 1,
      subject: "Projects",
      priority: "low",
      status: "new",
      description: "I am ok please call me now",
    },
    {
      id: 2,
      subject: "Development",
      priority: "high",
      status: "in progress",
      description: "I am ok please call me now",
    },

    {
      id: 3,
      subject: "Electricty",
      priority: "high",
      status: "new",
      description: "I am ok please call me now as i have a big problem",
    },
    {
      id: 4,
      subject: "Building",
      priority: "low",
      status: "done",
      description: "I am ok please call me now",
    },
  ];

  return (
    <section className="grid grid-cols-1 md:grid-cols-2 gap-6">
      {list.map((item) => (
        <TicketCard key={item.id} card={item} />
      ))}
    </section>
  );
};

export default TickectsList;
