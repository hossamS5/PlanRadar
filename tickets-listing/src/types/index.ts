export type ListCard = {
  id: number;
  subject: string;
  priority: string;
  status: string;
  description: string;
};
