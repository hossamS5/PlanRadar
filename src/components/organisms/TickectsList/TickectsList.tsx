import React from "react";
import TicketCard from "../../molecules/TicketCard/TicketCard";
import { ListCard } from "../../../types";

interface IProps {
  tickets: ListCard[];
}

const TickectsList: React.FC<IProps> = ({ tickets }) => {
  if (tickets?.length === 0) return <p>No tickets available</p>;

  return (
    <section className="grid grid-cols-1 md:grid-cols-2 gap-6">
      {tickets.map((item) => (
        <TicketCard key={item.id} card={item} />
      ))}
    </section>
  );
};

export default TickectsList;
