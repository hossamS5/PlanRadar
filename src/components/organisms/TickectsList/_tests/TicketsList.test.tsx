import { render, screen } from "@testing-library/react";
import TickectsList from "../TickectsList";
import { ListCard } from "../../../../types";

describe("TicketsList", () => {
  it("should render no tickets when tickets array is empty", () => {
    render(<TickectsList tickets={[]} />);

    expect(screen.getByText(/No tickets/i)).toBeInTheDocument();
  });

  it("matches snapshot", () => {
    const tickets: ListCard[] = [
      {
        id: 1,
        subject: "Projects",
        priority: "low",
        status: "new",
        description: "I am ok please call me now",
      },
      {
        id: 2,
        subject: "Development",
        priority: "high",
        status: "in progress",
        description: "I am ok please call me now",
      },
    ];
    const { asFragment } = render(<TickectsList tickets={tickets} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
