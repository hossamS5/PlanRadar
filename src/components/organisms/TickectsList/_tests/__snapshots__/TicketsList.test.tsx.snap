// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TicketsList > matches snapshot 1`] = `
<DocumentFragment>
  <section
    class="grid grid-cols-1 gap-6 md:grid-cols-2"
  >
    <div
      class="w-full p-4 rounded-[20px] border border-grayPalette-300 flex flex-col gap-4 shadow-md cursor-pointer"
      data-testid="ticket-card"
    >
      <div
        class="flex items-center justify-between gap-3"
      >
        <h2
          class="text-mainTheme text-xl font-bold"
        >
          Projects
        </h2>
        <span
          class="p-1 flex items-center bg-primary-300 text-white rounded-lg text-xs"
        >
          new
        </span>
      </div>
      <div
        class="text-[#464646]"
      >
        I am ok please call me now
      </div>
      <div
        class="text-xs bg-primary-800 text-white rounded-lg p-[6px] w-fit"
      >
        low
      </div>
    </div>
    <div
      class="w-full p-4 rounded-[20px] border border-grayPalette-300 flex flex-col gap-4 shadow-md cursor-pointer"
      data-testid="ticket-card"
    >
      <div
        class="flex items-center justify-between gap-3"
      >
        <h2
          class="text-mainTheme text-xl font-bold"
        >
          Development
        </h2>
        <span
          class="p-1 flex items-center bg-primary-300 text-white rounded-lg text-xs"
        >
          in progress
        </span>
      </div>
      <div
        class="text-[#464646]"
      >
        I am ok please call me now
      </div>
      <div
        class="text-xs bg-primary-800 text-white rounded-lg p-[6px] w-fit"
      >
        high
      </div>
    </div>
  </section>
</DocumentFragment>
`;
